raiden/utils/gevent_utils.py

```diff
@@ -1,30 +1,89 @@
 import gevent
 
+from raiden.exceptions import UnhandledExceptionInGreenlet
 
-def _make_greenlet_error_handler(original_handler):
-    def handle_error(self, context, type, value, tb):
-        if not issubclass(type, self.NOT_ERROR):
+
+class RaidenGreenlet(gevent.Greenlet):
```

Comment:
Can you add a comment here explaining what this is doing and especially how the `_wrap_link()` works. Took me some time to decipher so a comment here would be appreciated.

You have a very nice [comment below](https://github.com/raiden-network/raiden/pull/2041/files#diff-531ec925588de65e6d9f6e6c185083fdR46) but the one I am asking is more about how this class does what it does.